superset/charts/api.py

```diff
@@ -637,9 +632,7 @@ def get_data(self, pk: int) -> Response:
         except (TypeError, json.decoder.JSONDecodeError):
             form_data = {}
 
-        return self.get_data_response(
-            command, viz_type=chart.viz_type, form_data=form_data
```

Comment:
`viz_type` is also a field in `form_data`, so we just need to pass `form_data` around.